tools/kops_local_ccm.sh

```diff
@@ -0,0 +1,171 @@
+#!/usr/bin/env bash
+
+# Copyright 2022 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+set -x
+
+# This script brings up a Kubernetes cluster using kops and a local CCM image.
+# It is based on the e2e test script in e2e/scenarios/kops-simple.
+
+REPO_ROOT=$(git rev-parse --show-toplevel)
+cd ${REPO_ROOT}
+cd ..
+WORKSPACE=$(pwd)
+
+# Create bindir
+BINDIR=${WORKSPACE}/bin
+export PATH=${BINDIR}:${PATH}
+mkdir -p "${BINDIR}"
+
+# Setup our cleanup function; as we allocate resources we set a variable to indicate they should be cleaned up
+function cleanup {
+  if [[ "${DELETE_CLUSTER:-}" == "true" ]]; then
+      kops delete cluster --name "${CLUSTER_NAME}" --state "${KOPS_STATE_STORE}" --yes || echo "kops delete cluster failed"
+  fi
+}
+trap cleanup EXIT
+
+# Default cluster name
+SCRIPT_NAME=$(basename $0 .sh)
+if [[ -z "${CLUSTER_NAME:-}" ]]; then
+  CLUSTER_NAME="${SCRIPT_NAME}.k8s.local"
+fi
+echo "CLUSTER_NAME=${CLUSTER_NAME}"
+
+# Default workdir
+if [[ -z "${WORKDIR:-}" ]]; then
+  WORKDIR="${WORKSPACE}/clusters/${CLUSTER_NAME}"
+fi
+mkdir -p "${WORKDIR}"
+
+if [[ -z "${GCP_PROJECT:-}" ]]; then
+  echo "GCP_PROJECT must be set"
+  exit 1
```

Comment:
Can we move the requirements checks closer to the beginning? Seems like we should do this first, then start taking actions.